moto/sagemaker/responses.py

```diff
@@ -796,3 +796,174 @@ def update_endpoint_weights_and_capacities(self) -> TYPE_RESPONSE:
             desired_weights_and_capacities=desired_weights_and_capacities,
         )
         return 200, {}, json.dumps({"EndpointArn": endpoint_arn})
+
+    def list_model_packages(self):
+        creation_time_after = self._get_param("CreationTimeAfter")
+        creation_time_before = self._get_param("CreationTimeBefore")
+        max_results = self._get_param("MaxResults")
+        name_contains = self._get_param("NameContains")
+        model_approval_status = self._get_param("ModelApprovalStatus")
+        model_package_group_name = self._get_param("ModelPackageGroupName")
+        model_package_type = self._get_param("ModelPackageType")
+        next_token = self._get_param("NextToken")
+        sort_by = self._get_param("SortBy")
+        sort_order = self._get_param("SortOrder")
+        (
+            model_package_summary_list,
+            next_token,
+        ) = self.sagemaker_backend.list_model_packages(
+            creation_time_after=creation_time_after,
+            creation_time_before=creation_time_before,
+            max_results=max_results,
+            name_contains=name_contains,
+            model_approval_status=model_approval_status,
+            model_package_group_name=model_package_group_name,
+            model_package_type=model_package_type,
+            next_token=next_token,
+            sort_by=sort_by,
+            sort_order=sort_order,
+        )
+        # TODO: adjust response
+        response_values = [
+            "ModelPackageName",
+            "ModelPackageGroupName",
+            "ModelPackageVersion",
+            "ModelPackageArn",
+            "ModelPackageDescription",
+            "CreationTime",
+            "ModelPackageStatus",
+            "ModelApprovalStatus",
+        ]
+        model_package_summary_list_response_object = [
+            x.gen_response_object() for x in model_package_summary_list
+        ]
+        model_package_summary_list_response_object_filtered = [
+            {key: x[key] for key in response_values}
+            for x in model_package_summary_list_response_object
+        ]
+        for response_object in model_package_summary_list_response_object_filtered:
+            response_object["CreationTime"] = response_object["CreationTime"].isoformat()
```

Comment:
Similar point - the `ModelPackage` should be responsible for exposing the exact format of all it's fields.

The list-method needs `CreationTime` as a `datetime`-object, and we need it here in a specific format - so we could just define both attributes inside `ModelPackage`:
```
self._creation_time: datetime = ...  # used in the list-logic
self.creation_time = self._creation_time.isoformat()  # automatically exposed via `gen_response_object()`
```

Reply:
https://github.com/getmoto/moto/pull/6454/commits/d03894aad33a2c2f74f0bb74e64076618dcd384f

Reply:
I didnt move the model_package_version logic to the class because I need the backend to calculate the model_package_version itself. Moving this logic to the class would mean copying the list of modelpackages in each class which I think is not very sexy

Reply:
The problem might be solved by solving previous comments. Please could you check and let me know ?